pennylane/qinfo/transforms.py

```diff
@@ -58,11 +58,14 @@ def wrapper(*args, **kwargs):
         if len(measurements) != 1 or not isinstance(measurements[0], StateMP):
             raise ValueError("The qfunc measurement needs to be State.")
 
+        dm_measurement = measurements[0].wires or "mixed" in qnode.device.name
```

Comment:
Separating `StateMP` from `DensityMatrixMP` would definitely affect this line.

It'd be especially handy if `StateMP` always returned a state and a `DensityMatrixMP` always returned a density matrix without being device dependent.

Reply:
Yep, I think we'd all prefer that. Since the separation is in prototyping right now, I assume it won't be merged anytime soon?  I can add a TODO here mentioning that this would need updating once it's done.